### PR TITLE
Use config keys as primary in documentation, rename [Packages] to [Content]

### PR DESCRIPTION
**Pull request by keszybz**
This implements the first part of #769. While working on this, I found a bunch of small discrepancies, and subsequent commits fix a bunch. See individual commit messages for details.

(I had to merge #768, since this PR builds on it.)

### CODE REVIEW

File: mkosi.md [hunk 1]
```diff
@@ -380,61 +392,66 @@ details see the table below.
...
+  cache files are removed, too, before the re-build is initiated. If a
+  package cache is used (also see the "Files" section below),
+  specifying this option thrice will ensure the package cache is
+  removed too, before the re-build is initiated. For the `clean`
+  operation this option has a slightly different effect: by default
+  the verb will only remove build artifacts from a previous run, when
+  specified once the incremental cache files are deleted too, and when
+  specified twice the package cache is also removed.
+
+  <!--  FIXME: allow `Force=<n>` -->
```
Comment (DaanDeMeyer) [inline mkosi.md:429]: I'm wondering whether we should do Force in config files at all. This seems like one of those options that's better off as a CLI only option. Same with --incremental.

General rant: We should get rid of the argparse hack and generate the config parser and argparse object from the `CommandLineArguments` dataclass we have now (which we should rename to just `Args` or something similar.

File: mkosi.md [hunk 2]
```diff
@@ -797,183 +804,258 @@ details see the table below.
...
+`Autologin=`, `--autologin`
 
-: Enable autologin for the `root` user on pts/0 (nspawn), tty1 (QEMU) and
-  ttyS0 (QEMU with `--qemu-headless`) by patching `/etc/pam.d/login`.
+: Enable autologin for the `root` user on `/dev/pts/0` (nspawn),
+  `/dev/tty1` (QEMU) and `/dev/ttyS0` (QEMU with `QemuHeadless=yes`)
+  by patching `/etc/pam.d/login`.
 
-`--extra-search-paths=`
+<!-- FIXME: why are all those settings in [Validation] ?! -->
```
Comment (DaanDeMeyer) [inline mkosi.md]: The groupings don't really make a lot of sense for a lot of settings. AFAIK we don't even enforce them. I wonder if we should drop them alltogether.

**Comment by DaanDeMeyer** [comment]
DIdn't do a super detailed look through every change but LGTM.

File: mkosi.md (hunk 2 above)
Comment (keszybz) [reply mkosi.md]: I think they are useful to make it easier for humans to read the config files… I'd keep them, at least for now.

**Comment by keszybz** [comment]
The error from CI is pyright complaining about str↔Path assignments. I have a patch ready to rework most of that code, so I don't think it makes sense to fix the issue here. (It's not related to this PR, but preexisting.)

OK, since this is mostly moving documentation text around and @DaanDeMeyer gave it a thumbs up, let's merge.

File: mkosi/__init__.py [hunk 3]
```diff
@@ -6142,6 +6144,9 @@ def print_summary(args: CommandLineArguments) -> None:
     if args.build_script:
         MkosiPrinter.info("                 Run tests: " + yes_no(args.with_tests))
 
+    MkosiPrinter.info("                  Password: " + ("default" if args.password is None else "set"))
```
Comment (poettering) [inline mkosi/__init__.py:6147]: maybe actually change left hand side to "Root Password"?

File: mkosi.md (hunk 1 above)
Comment (poettering) [reply mkosi.md:429]: > I'm wondering whether we should do Force in config files at all. This seems like one of those options that's better off as a CLI only option. Same with --incremental.

I agree, Force in the config file is weird. Maybe at least drop from the docs, and keep it in the code for compat?

**Pull request merged**